test/Autofac.Specification.Test/Util/AsyncOnlyDisposeTracker.cs

```diff
@@ -28,6 +28,7 @@ public async ValueTask DisposeAsync()
 
             IsAsyncDisposed = true;
             Disposing?.Invoke(this, EventArgs.Empty);
+            GC.SuppressFinalize(this);
```

Comment:
Feels weird to add performance optimisation calls to a test-only implementation of a dispose tracker, no? Are we sacrificing test readability just to satisfy the analyser?

I have in the past added a `TestSuppressions.cs` file to test projects where I want to add assembly-level suppressions without putting them in the editorconfig?

Reply:
We could have two different analyzer.ruleset files, one for production code, one for tests. I've done that for other projects.

I didn't think this really affected readability so much and... I mean, it _is_ disposing, so _technically_ it's correct, right?

I don't feel terribly strongly about it. It might be worth having two different ruleset files just to add the flexibility.

Reply:
I added `Test.ruleset` with some of the various suppressions that were around the test projects. It should clean up a bit of the `#pragma` and other suppression noise in there. Also removed the `GC` stuff here and added that rule to the ruleset so it won't be enforced in tests.